ui/apps/platform/cypress/integration/violations/filteredWorkflowViews.test.ts

```diff
@@ -0,0 +1,87 @@
+import withAuth from '../../helpers/basicAuth';
+import { hasFeatureFlag } from '../../helpers/features';
+import { visitViolations } from './Violations.helpers';
+
+const filteredWorkflowSelectButton = 'button[aria-label="Filtered workflow select"]';
+const getFilteredWorkflowOptionButton = (viewName) =>
+    `div[aria-label="Filtered workflow select options"] button:contains("${viewName}")`;
+
+describe('Violations - Filtered Workflow Views', () => {
+    withAuth();
+
+    before(function () {
+        if (!hasFeatureFlag('ROX_PLATFORM_COMPONENTS')) {
+            this.skip();
+        }
+    });
+
+    it('should render the correct filtered workflow view select options', () => {
+        visitViolations();
+
+        // should exist and display "Applications view" by default
+        cy.get(filteredWorkflowSelectButton).should('exist');
+        cy.get(filteredWorkflowSelectButton).should('have.text', 'Applications view');
+
+        // show the dropdown options
+        cy.get(filteredWorkflowSelectButton).click();
+
+        // the correct options should display in the dropdown
+        const options = ['Applications view', 'Platform view', 'Full view'];
+        options.forEach((option, index) => {
+            cy.get(
+                `div[aria-label="Filtered workflow select options"] li:nth(${index}) button span.pf-v5-c-menu__item-text`
+            ).should('have.text', option);
+        });
+    });
+
+    it('should filter the violations table when the "Applications view" is selected', () => {
+        visitViolations();
+
+        cy.intercept('GET', '/v1/alerts?query=*').as('getViolations');
+
+        // should filter using the correct values for the "Applications view"
```

Comment:
Nothing in the table or data will let me know if the result is a platform violation, application violation, etc. Because of that, I can't test using the table to check if the results are properly filtered. The next best option was to check if the API uses the right filters. I believe it's a good proxy for confirming the filter behavior.

Reply:
Yeah, I came to a similar conclusion for some tests in VM. Ideally we would test fully e2e, but given the constraints I think an integration test that asserts the correct parameters are sent is the next best option.

I think it is likely that Violations shares the same issue as VM too -- we can't really be sure what data, or how much data, will be returned during a CI run. Even if the result did distinguish between application and platform violations, there is a non-zero chance that there simply are no violations at the point that this test runs.